x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/types.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export type SchemaTypes = 'text' | 'number' | 'geolocation' | 'date';
+
+export interface Schema {
+  [key: string]: SchemaTypes;
+}
+
+// this is a mapping of schema field types ("string", "number", "geolocation", "date") to the names
+// of source engines which utilize that type
+export type SchemaConflictFieldTypes = {
+  [key in SchemaTypes]: string[];
+};
+
+export interface SchemaConflict {
+  fieldTypes: SchemaConflictFieldTypes;
+  resolution?: string;
+}
+
+// For now these values are ISchemaConflictFieldTypes, but in the near future will be ISchemaConflict
+// once we implement schema conflict resolution
```

Comment:
I migrated these comments as-is - not sure if they still apply, @byronhulcher might have more context

Reply:
I don't think we're implementing schema conflict resolution any time soon, so probably no longer relevant.

Reply:
👍 I'll take a closer look at removing when we migrate the schema view. Thanks Jason!